From f34b284d3f2b6cdeb08f247c16d5b7a69cb45061 Mon Sep 17 00:00:00 2001
From: Pavel Shirshov <pshirshov@eml.cc>
Date: Tue, 25 Oct 2022 22:01:55 +0100
Subject: [PATCH] fixes #559, #485: workaround for broken fontrestore on
 Ventura

---
 modules/fonts/default.nix | 30 ++++++++++++++++--------------
 1 file changed, 16 insertions(+), 14 deletions(-)

diff --git a/modules/fonts/default.nix b/modules/fonts/default.nix
index 14f9529f..44ea58bb 100644
--- a/modules/fonts/default.nix
+++ b/modules/fonts/default.nix
@@ -58,20 +58,22 @@ in
           fi
       done
 
-      fontrestore default -n 2>&1 | while read -r f; do
-          case $f in
-              /Library/Fonts/*)
-                  font=''${f##*/}
-                  if [ ! -e "$systemConfig/Library/Fonts/$font" ]; then
-                      echo "removing font $font..." >&2
-                      rm "/Library/Fonts/$font"
-                  fi
-                  ;;
-              /*)
-                  # ignoring unexpected fonts
-                  ;;
-          esac
-      done
+      if [[ "`sw_vers -productVersion`" < "13.0" ]]; then
+        fontrestore default -n 2>&1 | while read -r f; do
+            case $f in
+                /Library/Fonts/*)
+                    font=''${f##*/}
+                    if [ ! -e "$systemConfig/Library/Fonts/$font" ]; then
+                        echo "removing font $font..." >&2
+                        rm "/Library/Fonts/$font"
+                    fi
+                    ;;
+                /*)
+                    # ignoring unexpected fonts
+                    ;;
+            esac
+        done
+      fi
     '';
 
   };
